qark/plugins/Manifest_File_Checks.py

```diff
@@ -1,22 +1,71 @@
+    STRING = ("Data Injection due to exported Broadcast Receiver.\n"
+              "Default priority of exported receiver is 0. Since, the higher priority receivers respond first"
+              " and forward it to lower priority receivers, a malicious receiver with high priority can intercept the "
+              "message change it and forward it to lower priority receivers.\n{}\n")
+    return STRING.format(list_orderedBR)
+
+
+def google_safe_browsing(fileName):
+    STRING = ("To provide users with a safer browsing experience, you can configure your apps"
+              "WebView objects to verify URLs using Google Safe Browsing.\nWhen this security measure is enabled,"
+              " your app shows users a warning when they attempt to navigate to a potentially unsafe website.\n{}\n")
+    return STRING.format(fileName)
+
+
+def task_launch_mode(line):
+    STRING = ("Use of android:launchMode=singleTask is found\n"
+              "This results in AMS either resuming the earlier activity or loads it in a task with same affinity or"
+              " the activity is started as a new task. This may result in Task Poisoning.\n"
+              "https://www.usenix.org/system/files/conference/usenixsecurity15/sec15-paper-ren-chuangang.pdf\n{}\n")
+    return STRING.format(line)
 
+
+def task_reparenting(line):
+    STRING = ("Use of android:allowTaskReparenting='true' is found\n"
+              "This allows an existing activity to be reparented to a new native task"
+              " i.e task having the same affinity as the activity.\n"
+              "This may lead to UI spoofing attack on this application.\n"
+              "https://www.usenix.org/system/files/conference/usenixsecurity15/sec15-paper-ren-chuangang.pdf\n{}\n")
+    return STRING.format(line)
+
+
+def path_usage(line):
+    STRING = ("Be careful with the use of android:path in the <path-permission> tag\n"
+              "android:path means that the permission applies to the exact path declared"
+              " in android:path. This expression does not protect the sub-directories\n{}\n")
+    return STRING.format(line)
+
+
+def hardcoded_api_key(api_key_variable):
+    STRING = "API Key Found\n{}"
+    return STRING.format(api_key_variable)
+
+
+class ManifestFilePlugin(IPlugin):
     PATH_USAGE = r'android:path='
     # Regex to check the launch mode of activity and task
     LAUNCH_MODE = r'android:launchMode=[\'\"]singleTask[\'\"]'
     TASK_REPARENTING = r'android:allowTaskReparenting=[\'\"]true[\'\"]'
     RECEIVER_REGEX = r'<receiver.*?>'
     PRIORITY_REGEX = r'priority'
+    HARDCODED_API_KEY = r'API_KEY|api_key|API|api|key'
+    # Regex to check for API Keys containing atleast one uppercase, lowercase chracters and number
+    API_KEY_REGEX = r'^.+(?=.{20,})(?=.+\d)(?=.+[a-z])(?=.+[A-Z]).+$'
+    # Regex which is used to ignore few special characters in api keys to prevent false positives
+    SPECIAL_CHAR_REGEX = r'^.+(?=.+[!$%^~]).+$'
```

Comment:
Not all of your regex has to start with `^` and end with `$`. You can simplify this regex by actually just using `[!$%^~]` and not dealing with stuff thats before and after it. If you are only interested in those characters, then search for them and not other things.